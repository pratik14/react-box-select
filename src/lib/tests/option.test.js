import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Option from '../option';

describe('<Option />', () => {
  const option = { id: '1', label: 'One' };

  it('should render without crashing', () => {
    const tree = renderer
      .create(
        <Option
          type="selectable"
          option={option}
          text={option.label}
          handleClick={() => {}}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('#handleOptionClick', () => {
    const mockCallback = jest.fn();
    const wrapper = shallow(
      <Option
        type="selectable"
        option={option}
        text={option.label}
        handleClick={mockCallback}
      />,
    );

    wrapper.simulate('click');
    expect(mockCallback.mock.calls.length).toBe(1);
    expect(mockCallback).toBeCalledWith(option);
  });

  it('#disabled', () => {
    option.disabled = true;
    const tree = renderer
      .create(
        <Option
          type="selectable"
          option={option}
          text={option.label}
          handleClick={() => {}}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
