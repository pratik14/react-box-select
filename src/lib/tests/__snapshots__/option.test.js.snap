// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Option /> should render without crashing 1`] = `
<li
  className="ms-elem-selectable"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
>
  One
   
</li>
`;
